Radio.m

```diff
@@ -1445,14 +1591,66 @@ - (void) parseGpsToken: (NSScanner *) scan {
     
 }
 
-
+//
+// Parse Profile tokens
+//     called on the GCD thread associated with the GCD tcpSocketQueue
+//
+//     format: <apiHandle>|profile <profileType> list=<value>^<value>^...<value>^
+//                           OR
+//     format: <apiHandle>|profile <profileType> current=<value>
+//
+//     scan is initially at scanLocation = 17, start of the <profileType>
+//     "<apiHandle>|profile " has already been processed
+//
 - (void) parseProfileToken: (NSScanner *) scan {
-    
+    // get the Profile type
+    NSString *profileType;
+    [scan scanUpToString:@" " intoString: &profileType];
+    // skip the space
+    [scan scanString:@" " intoString: nil];
+    
+    // get the Sub type
+    NSString *profileSubType;
+    [scan scanUpToString:@"=" intoString: &profileSubType];
+    // skip the "="
+    [scan scanString:@"=" intoString: nil];
+    
+    // get the remainder of the command
+    NSString *remainderOfCommand;
+    [scan scanUpToString:@"\n" intoString: &remainderOfCommand];
+    
+    // List or Current value?
+    if ([profileSubType isEqualToString: @"list"]) {
+        // it's the List, separate the components
+        NSMutableArray *profileNames;
+        profileNames = [[remainderOfCommand componentsSeparatedByString:@"^"] mutableCopy];
+        // remove the last (empty) string
+        [profileNames removeLastObject];
+        // save it in the appropriate property
+        if ([profileType isEqualToString: @"global"]) {
+            @synchronized (self.globalProfiles) {
+                _globalProfiles = profileNames;
```

Comment:
I did find a curious item in the C# code. Whenever they Save or Delete a TxProfile they replace all "*" with "" as in:

[profile stringByReplacingOccurrencesOfString:@"*" withString:@""]

I've added this into the code.